### PR TITLE
Add: Add open-file-explorer definition

### PR DESCRIPTION
**Pull request by sevenc-nanashi**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/open-file-explorer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for open-file-explorer 1.0
+// Project: https://github.com/vibhor1997a/nodejs-open-file-explorer#readme
+// Definitions by: sevenc-nanashi <https://github.com/sevenc-nanashi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = open_file_explorer;
```
Comment (peterblazejewicz) [inline types/open-file-explorer/index.d.ts:6]:
```suggestion
export = openExplorer;
```

File: types/open-file-explorer/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+declare function open_file_explorer(path: string, callback: (err: Error | undefined) => void): void;
```
Comment (peterblazejewicz) [inline types/open-file-explorer/index.d.ts:8]:
```suggestion
declare function openExplorer(path: string, callback: (err: Error | undefined) => void): void;
```

File: types/open-file-explorer/open-file-explorer-tests.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import openExplorer = require('open-file-explorer');
+
+// $ExpectType void
+openExplorer('.', (err) => {
+    if (err) {
+        console.log(err);
```
Comment (peterblazejewicz) [inline types/open-file-explorer/open-file-explorer-tests.ts:6]:
```suggestion
        err; // $ExpectType Error
```

File: types/open-file-explorer/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
```
Comment (peterblazejewicz) [inline types/open-file-explorer/tsconfig.json:6]:
```suggestion
            "es6"
```
removing `console`, removes the need for a 'Dom' library as dependency

**Comment by sevenc-nanashi** [comment]
Thank you for your review! I've updated the PR.

**Review by peterblazejewicz — APPROVED** [review]
@sevenc-nanashi thanks!
LGTM!

**Comment by sevenc-nanashi** [comment]
Ready for merge

**Comment by sevenc-nanashi** [comment]
Ready to merge

**Pull request merged**